### PR TITLE
Let override even before onAfterRoute event

### PR DESCRIPTION
**Pull request by shandak**

### CODE REVIEW

File: mvcoverride.php [hunk 1]
```diff
@@ -58,92 +64,131 @@ public function __construct(&$subject, $config = array())
...
-		$app = JFactory::getApplication();
-
-		$includePath = $this->params->get('includePath', '{JPATH_BASE}/code,{JPATH_THEMES}/{template}/code');
-
-		$includePath = str_replace(
-			array('{JPATH_BASE}', '{JPATH_THEMES}', '{template}'),
-			array(JPATH_BASE, JPATH_THEMES, $app->getTemplate()),
-			$includePath
-		);
+		$redcoreLoader = JPATH_LIBRARIES . '/redcore/bootstrap.php';
```
Comment (jatitoam) [inline mvcoverride.php]: I recall there was a reason why all the loading was not done in the constructor before, do you remember any of that?

File: mvcoverride.php [hunk 2]
```diff
@@ -309,42 +368,4 @@ private function getOverrideFileInfo($includePath, $component, $filePath, $type
...
 		);
 	}
-
-	/**
-	 * Get option
-	 *
-	 * @return boolean|mixed|string
-	 * @since  1.4
-	 */
-	private function getOption()
```
Comment (jatitoam) [inline mvcoverride.php:319]: Cleaning up?

File: mvcoverride.php (hunk 1 above)
Comment (shandak) [reply mvcoverride.php]: yes, I remember that and I built a double initialisation:
First right after the plugin will be loaded so we can use it everywhere  and not the only afterRoute
Second when onAfterRoute will be executed, so if someone still needs to use a template folder for overrides or add new pathways via other plugins - they can do it, but as it was before - only in onAfterRoute

File: mvcoverride.php (hunk 2 above)
Comment (shandak) [reply mvcoverride.php:319]: yep, it's not useful code.

File: mvcoverride.php (hunk 1 above)
Comment (shandak) [reply mvcoverride.php]: The main problem was - we can't get the current template if we did not pass the router where we need to check Itemid.
Now I split the root override path and template override path. Now they have a separate initialization.

**Review by jatitoam — APPROVED** [review]

**Pull request merged**